### PR TITLE
fix #459: using .scalar_subquery() for SQLA>1.4

### LINKED ISSUE #459: SAWarning: implicitly coercing SELECT object to scalar subquery
when running pytest getting scalar_subquery warning in `tests/conftest.py` with SQLA > 1.4
```
mashmallow-sqlalchemy/tests/conftest.py:102: SAWarning: implicitly coercing SELECT object to scalar subquery; please use the .scalar_subquery() method to produce a scalar subquery.
    course_count = column_property(

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

### PR DESCRIPTION
**Pull request by indiVar0508**
Attempt to address #459 ,

Using scalar_subquery available from SQLA>1.4

### CODE REVIEW

**Comment by indiVar0508** [comment]
Hi,

Not sure but the jobs are failing with below shown error, can you help me in addressing the [issue](https://dev.azure.com/sloria/sloria/_build/results?buildId=12101&view=logs&j=ea246cef-d1aa-57f5-7a3e-96f81afd1e8b&t=dbea0cb8-2fce-5953-3cb5-dc5f7d114bd9&l=49)?

`ERROR: InvocationError for command /home/vsts/work/1/s/.tox/lint/bin/pre-commit run --all-files (exited with code 1)`

**Comment by lafrech** [comment]
Install pre-commit in your local env to run black locally (see CONTRIBUTING.rst).

**Comment by indiVar0508** [comment]
thanks @lafrech , it worked now, it was some formatting issue.

File: tests/conftest.py [hunk 1]
```diff
@@ -98,12 +98,20 @@ class Student(Base):
             secondary=student_course,
             backref=backref("students", lazy="dynamic"),
         )
-        # Test complex column property
-        course_count = column_property(
-            sa.select([sa.func.count(student_course.c.course_id)]).where(
-                student_course.c.student_id == id
+        try:
+            subquery = (
```
Comment (AbdealiLoKo) [inline tests/conftest.py]: Looks good - can we avoid repeating:
```
sa.select([sa.func.count(student_course.c.course_id)])
.where(student_course.c.student_id == id)
```
and just write that once on the top ?

**Comment by lafrech** [comment]
Thanks.

For the record, the change is documented [here](https://docs.sqlalchemy.org/en/14/changelog/changelog_14.html#change-f413ff8c13a1fb7d766bfa16dfcdfbf1). This is the right fix. Once @AbdealiJK's comment about factoring the subquery is addressed, this can be merged.

File: tests/conftest.py (hunk 1 above)
Comment (indiVar0508) [reply tests/conftest.py]: Hi @AbdealiJK ,

Made the suggested changes

File: tests/conftest.py [hunk 2]
```diff
@@ -98,12 +98,16 @@ class Student(Base):
...
-            )
+        subquery = sa.select([sa.func.count(student_course.c.course_id)]).where(
+            student_course.c.student_id == id
         )
+        if hasattr(subquery, "scalar_subquery"):
+            subquery = subquery.scalar_subquery()
+        else:  # SQLA < 1.4
+            subquery = subquery.as_scalar()
+
+        # Test complex column property
```
Comment (lafrech) [inline tests/conftest.py]: Nitpick: I'd leave that comment above the `subquery =` line, and remove the empty line (l. 108).

**Review by lafrech — APPROVED** [review]
Good to me. I'd just move that comment line.

File: tests/conftest.py (hunk 2 above)
Comment (indiVar0508) [reply tests/conftest.py]: ok sure, should i update like this?
```
        if hasattr(subquery, "scalar_subquery"):
            subquery = subquery.scalar_subquery()
        else:  
            # SQLA < 1.4
            subquery = subquery.as_scalar()
        # Test complex column property
        ...
```
is this ok?

Comment (lafrech) [reply tests/conftest.py]: I meant the comment applies to the whole thing.

```py
        # Test complex column property
        subquery = sa.select([sa.func.count(student_course.c.course_id)]).where(
            student_course.c.student_id == id
        )
        if hasattr(subquery, "scalar_subquery"):
            subquery = subquery.scalar_subquery()
        else:  # SQLA < 1.4
            subquery = subquery.as_scalar()
	course_count = column_property(subquery)
```

Again, sorry about this, just a detail.

Comment (indiVar0508) [reply tests/conftest.py]: ohh ok got it, updating the changes

**Comment by lafrech** [comment]
Thanks !

**Pull request merged**